code/modules/reagents/newchem/food.dm

```diff
@@ -133,17 +133,19 @@ datum/reagent/honey/reaction_turf(var/turf/T, var/volume)
 	reagent_state = LIQUID
 	color = "#2E2418"
 
-datum/reagent/chocolate/on_mob_life(var/mob/living/M as mob)
+/datum/reagent/chocolate/on_mob_life(var/mob/living/M as mob)
 	if(!M) M = holder.my_atom
 	M.reagents.add_reagent("sugar", 0.8)
 	..()
 	return
 
-datum/reagent/chocolate/reaction_turf(var/turf/T, var/volume)
+/datum/reagent/chocolate/reaction_turf(var/turf/T, var/volume)
 	src = null
-	if(volume >= 5)
-		new /obj/item/weapon/reagent_containers/food/snacks/reagentchocolatebar(T)
-		return
+	var/volume_left = volume
+	while(volume_left >= 5)
```

Comment:
@FalseIncarnate 

DZD said he has no real opinion on this. 

I'd just leave the behavior as it was with your cocoa pile changes.